server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java

```diff
@@ -75,7 +86,11 @@ public MIMEMessageConverter() {
         }
 
         MessageBuilder messageBuilder = MessageBuilder.create();
-        messageBuilder.setBody(createTextBody(creationMessageEntry.getMessage()));
+        if (mixedTextAndHtml(creationMessageEntry.getMessage())) {
+            messageBuilder.setBody(createMultipartBody(creationMessageEntry.getMessage()));
```

Comment:
Be careful, all is here: when calling .setBody() with a multipart argument, the content type is correctly generated with boundaries. If you call it with a generic method which will answer Body instead of Multipart, the bad .setBody() is called and the content type is not complete.